pkg/utils/elasticsearch7.go

```diff
@@ -192,6 +192,14 @@ func (es *Elasticsearch7) CreateOrUpdateIndex(ctx context.Context, indexName str
 	return BuildEsStatus(response.StatusCode, response.String()), nil
 }
 
+// ptrToString return (nil) if the ptr is nil or the value
+func ptrToString(ptr *int32) string {
+	if ptr == nil {
+		return "(nil)"
+	}
+	return fmt.Sprintf("%d", *ptr)
+}
+
```

Comment:
You should move this code to `elasticsearch_common.go`, and add the same workaround to `elasticsearch8.go` too (for elasticsearch 8 client).

Reply:
When looking the code, there is lot of copy/paste. In my opinion, there is not only this part to move in "common stuff".

Reply:
We should work on the naming as well. 

`isShardsUpdated` if true it is in error with we cannot update.